app/routes/posts/index.tsx

```diff
@@ -0,0 +1,40 @@
+import { json } from "@remix-run/node";
+import { Link, useLoaderData } from "@remix-run/react";
+
+import { getPosts } from "~/models/post.server";
+
+type Post = {
+  slug: string;
+  title: string;
+};
```

Comment:
If this type is shared across multiple files you put the type definition in the file it makes the most sense (probably post.server?) and export it so you can import here

```
// post.server
export type Post {}
```

```
// posts/index.tsx
import { type Post } from "etc"
```